src/business-logic/services/tv2-inews-ingest-service.ts

```diff
@@ -1,11 +1,18 @@
 import { IngestService } from './interfaces/ingest-service'
-import {HttpRequestParameters, HttpService} from './interfaces/http-service'
+import { HttpRequestParameters, HttpService } from './interfaces/http-service'
+import { RundownRepository } from '../../data-access/repositories/interfaces/rundown-repository'
+import { Rundown } from '../../model/entities/rundown'
 
-export class Tv2InewsIngestService implements IngestService {
-  constructor(private readonly httpService: HttpService) {}
-  public async reloadIngestData(rundownName: string): Promise<void> {
-    const domain = process.env.INEWS_DOMAIN ?? 'localhost:3007'
-    const url = `http://${domain}/reloadData/${rundownName}`
+const DEFAULT_URL: string = 'localhost:3007'
+
+export class Tv2INewsIngestService implements IngestService {
+
+  constructor(private readonly httpService: HttpService, private readonly rundownRepository: RundownRepository) {}
+
+  public async reloadIngestData(rundownId: string): Promise<void> {
```

Comment:
This method is getting a bit "clamped".
You are also technically mixing abstractions levels.
A method called `getIngestUrl()` would help it tremendiously.
```
public async reloadIngestData(...): ... {
  const rundown: Rundown = await this.rundownRepository.getRundown(...)
  const url: string = this.getIngestUrl()
  const parameters: HttpRequestParameters = { url: url }
  await this.httpService.post(parameters)
}
```

Reply:
I have moved it to a private method! Good suggestion.